src/app.rs

```diff
@@ -175,6 +177,50 @@ pub struct Config {
     pub using_table: Option<String>,
     pub using_port: Option<u32>,
     // pub cache_expiration_time: Option<i64>, // in second. default 300 (= 5 minutes)
+    pub retry: Option<RetryConfig>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct RetryConfig {
+    pub default: RetrySetting,
+    pub batch_write_item: Option<RetrySetting>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct RetrySetting {
+    pub initial_backoff: Option<Duration>,
+    pub max_backoff: Option<Duration>,
+    pub max_attempts: Option<u32>,
+}
+
+impl Default for RetrySetting {
+    fn default() -> Self {
+        Self {
+            initial_backoff: None,
+            max_backoff: None,
+            max_attempts: Some(10),
+        }
+    }
+}
+
+impl From<RetrySetting> for ExponentialBuilder {
+    fn from(value: RetrySetting) -> Self {
+        let mut builder = Self::default()
+            .with_jitter()
+            .with_factor(2.0)
+            .with_min_delay(Duration::from_secs(1));
+
+        if let Some(max_attempts) = value.max_attempts {
+            builder = builder.with_max_times(max_attempts as usize);
```

Comment:
Based on my investigation, backon's `max_times` does not include the initial attempt. Thus, to maintain compatibility, we need to verify that its value is greater than zero and subtract one when passing it to backon.

Reply:
I fixed it.